src/Processors/Formats/Impl/ArrowBufferedStreams.cpp

```diff
@@ -41,9 +42,18 @@ arrow::Result<int64_t> ArrowBufferedOutputStream::Tell() const
 
 arrow::Status ArrowBufferedOutputStream::Write(const void * data, int64_t length)
 {
-    out.write(reinterpret_cast<const char *>(data), length);
-    total_length += length;
-    return arrow::Status::OK();
+    try
+    {
+        out.write(reinterpret_cast<const char *>(data), length);
+        total_length += length;
+        return arrow::Status::OK();
+    }
+    catch (...)
+    {
+        auto message = getCurrentExceptionMessage(false);
```

Comment:
May be it make sense to destroy `out` here with exception context.
`out` should not be used after exception any more.

Reply:
We hold only reference to this buffer, it wouldn't be destroyed in case of an exception previously. We finalize these buffers in corresponding StorageSink in `onException` methods, and we handle excetpion context there. For example:
https://github.com/ClickHouse/ClickHouse/blob/04ea8a85f21c36640ecc36d8327557e74e880375/src/Storages/StorageFile.cpp#L1788-L1800

Here exception should be thrown later from arrow code when it's processed

Reply:
I see that there is exception `Error while writing a table: {}` when `WriteTable` returns not ok.
If that status is checked always, that it is good and my proposal is superfluous.